app/controllers/publishers/jobseeker_profiles_controller.rb

```diff
@@ -1,12 +1,16 @@
 class Publishers::JobseekerProfilesController < Publishers::BaseController
   def index
-    @pagy, @jobseeker_profiles = pagy(JobseekerProfileQuery.new(params, current_organisation).call)
-    @form = Publishers::JobseekerProfilesForm.new(jobseeker_profiles_params)
+    @pagy, @jobseeker_profiles = pagy(Search::JobseekerProfileSearch.new(filters, current_organisation).jobseeker_profiles)
+    @form = Publishers::JobseekerProfileSearchForm.new(jobseeker_profile_search_params)
   end
 
   private
 
-  def jobseeker_profiles_params
+  def jobseeker_profile_search_params
     params.permit(qualified_teacher_status: [], roles: [], working_patterns: [], key_stages: [], education_phases: [])
   end
+
+  def filters
+    jobseeker_profile_search_params.each { |_, v| v.compact_blank! }
```

Comment:
This could perhaps be
```suggestion
    jobseeker_profile_search_params.transform_values(&:compact_blank)
```